crates/ra_hir/src/ty/infer/coerce.rs

```diff
@@ -0,0 +1,334 @@
+//! Coercion logic.
```

Comment:
Might be a good occasion to tell a bit about what coercion is, and when it is expected to happen